lightningd/pay.c

```diff
@@ -1177,6 +1178,14 @@ send_payment(struct lightningd *ld,
 		ret = pubkey_from_node_id(&pubkey, &ids[i]);
 		assert(ret);
 
+		if (dev_legacy_hop && i == n_hops - 2) {
+			sphinx_add_v0_hop(path, &pubkey,
+					  &route[i + 1].scid,
+					  route[i + 1].amount,
+					  base_expiry + route[i + 1].delay);
+			continue;
```

Comment:
Just a stylistic thing, are you avoiding `if {} else {}` for legacy and non-legacy to minimize the diff when we remove legacy again?